packages/runtime/container-runtime/src/gcSweepReadyUsageDetection.ts

```diff
@@ -74,16 +74,8 @@ export class SweepReadyUsageDetectionHandler {
         localStorageOverride?: Pick<Storage, "getItem" | "setItem">,
     ) {
         const noopStorage = { getItem: () => null, setItem: () => {} };
-        if (localStorageOverride !== undefined) {
-            this.localStorage = localStorageOverride;
-        } else {
-            try {
-                // localStorage is not defined in Node environment so this throws
-                this.localStorage = localStorage ?? noopStorage;
-            } catch (error) {
-                this.localStorage = noopStorage;
-            }
-        }
+        // localStorage is not defined in Node environment, so fall back to noopStorage if needed.
+        this.localStorage = localStorageOverride ?? globalThis.localStorage ?? noopStorage;
```

Comment:
Are you sure it's equivalent? I believe some browser environments have policies that can be enabled to not expose localStorage. And I think exception is fired when you try to access it. I might be mistaken, but I remember it's a minefield and we had to back out similar change in the past after it shipped to prod

Reply:
@christiango might have more insights

Reply:
There are some scenarios where local storage can be disabled by the user to avoid tracking and can result in an exception when accessed

Reply:
Do you remember when exception fires? When using it (i.e. calling get/set), or when retrieving it from window object?
@CraigMacomber, please take a look at https://stackoverflow.com/questions/17882647/can-user-disable-html5-sessionstorage#:~:text=In%20Chrome%3A%20Open%20%E2%80%9COptions%E2%80%9D%20and%20select%20%E2%80%9CUnder%20the,some%20browser%20security%20plugins%2Fextras%2Fadd-ons%20that%20will%20prevent%20localStorage.

Reply:
the exception is thrown when you access a global that does not exist by just using its name unqualified (ex: running the old code in node). If you access globals as members of globalThis (or the window object: they are the same in browsers), you get undefined instead of an exception.

Interestingly I just tested in firefox and it returns null instead of undefined when explicitly disabled regardless of which way its accessed. This should work fine in my code and the old code, as both fall back to noopStorage by using ?? in that case.

Reply:
As long as you tested those scenarios, I'm good with it:)